pkg/app/server/grpcapi/web_api.go

```diff
@@ -380,6 +380,34 @@ func (a *WebAPI) UpdatePipedDesiredVersion(ctx context.Context, req *webservice.
 	return &webservice.UpdatePipedDesiredVersionResponse{}, nil
 }
 
+func (a *WebAPI) RestartPiped(ctx context.Context, req *webservice.RestartPipedRequest) (*webservice.RestartPipedResponse, error) {
+	claims, err := rpcauth.ExtractClaims(ctx)
+	if err != nil {
+		a.logger.Error("failed to authenticate the current user", zap.Error(err))
+		return nil, err
+	}
+
+	piped, err := getPiped(ctx, a.pipedStore, req.PipedId, a.logger)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
should add check to ensure that the going to restart piped belongs to the user's project.
ref: https://github.com/pipe-cd/pipecd/blob/master/pkg/app/server/grpcapi/web_api.go#L653-L655

Reply:
I forgot it :bow:
Added check: https://github.com/pipe-cd/pipecd/pull/3725/commits/228221792b43fd44a0a66561a5c1128b03999625